docs/REST API/rest-api.md

```diff
@@ -3630,6 +3643,79 @@ Retrieves a list of the configuration profiles assigned to a host.
 }
 ```
 
+### Lock host
+
+_Available in Fleet Premium_
+
+Sends a command to lock the specified macOS, Linux, or Windows host. The host is locked once it comes online.
+
+`GET /api/v1/fleet/hosts/:id/lock`
+
+#### Parameters
+
+| Name       | Type              | In   | Description                                                                   |
+| ---------- | ----------------- | ---- | ----------------------------------------------------------------------------- |
+| id | integer | path | **Required**. ID of the host to be locked. |
+
+#### Example
+
+`GET /api/v1/fleet/hosts/123/lock`
+
+##### Default response
+
+`Status: 200`
+
+Requests to lock macOS hosts will also include the following response data:
+```json
+{
+  "host_id": 8,
+  "unlock_pin": 123456
```

Comment:
Unlock pin should be `null` if the host is windows or linux

Reply:
cc @noahtalerman — I remember in the design reviews you advocated for omitting this data from the response for Windows/Linux hosts, want to make sure you're on board